### ISSUE TITLE
Deep Neural Networks for Object Detection

### ISSUE THREAD

**Issue by jskim19** [GDSC-Ewha-4th/Study-paperReview#64]
### Abstract 
DNN은 최근 이미지 분류에서 뛰어난 성능을 보여줬다. 이 논문에서는 DNN을 이용해 객체를 분류하는 것 뿐만 아니라 다양한 클래스에서 객체를 localizing한다. 적은 네트워크 적용으로 낮은 비용으로 높은 해상도의 객체 탐지가 가능해진다. 

### Introduction 

- 이미지 내에서 객체의 클래스와 위치를 추정하는 것이 object detection 이라는 문제이며, 주요 발전은 object representations과 머신 러닝 모델로 이루어졌다. 

- 이 논문에서는 이미지에 여러 개의 객체가 있을 때, 그 경계 박스를 예측할 수 있는 공식을 보여준다. 또한 loalization의 정밀도를 높이기 위해 full image에 DNN mask generation을 적용한다. 

![IMG_1858](https://user-images.githubusercontent.com/113578083/204100004-47957ca2-0d87-4841-ac3b-43ff8a6d1ac2.jpg)

- DNN-based regression은 classification 뿐만 아니라 강력한 geometric information을 캡쳐할 수 있다. 

- classification을 위한 일반적인 구조를 사용하고 마지막 레이어만 regression layer로 대체하면 객체의 위치도 캡쳐할 수 있게 한다. 

- 또한 정밀한 detection을 위한 multi-scale box inference를 소개한다. 

### DNN-based Detection 

- 객체 마스크에 대한 DNN-based regression 이 핵심이다. 이 모델을 기반으로 객체 일부 뿐만 아니라 전체 객체에 대한 마스크를 만들 수 있다. localization의 정밀도를 높이기 위해 DNN localizer를 적용한다. 

### Detection as DNN Regression 

- 7 layer로 구성된, convolutional DNN을 기반으로 하는 네트워크를 사용한다. (앞 5개는 convolutional, 뒤 2개는 fully-connected)

- localization을 위해 일반적인 구조를 사용한다. 마지막 레이어에서 softmax classifier를 사용하는 대신 객체 이진 마스크를 만드는 regression layer를 사용한다. 

- 네트워크가 highly non-convex하고 최적임을 보장할 수 없기 때문에 loss function을 regularize해야한다. 

- 대부분의 객체가 이미지 사이즈에 비해 작다는 직관은 모든 아웃풋에 0을 내보내는 trivial한 솔루션을 내보낼 수 있다. 이런 바람직하지 않은 현상을 피하기 위해 ground truth mask의 non zero value의 가중치를 변수에 따라 증가시켜 준다. 

### Precise Object Localization via DNN-generated Masks 

- 하나의 객체 마스크는 서로 옆에 있는 객체를 명확하게 구분하기에 충분하지 않고, 아웃풋 크기의 한계로, 원본 이미지보다 훨씬 작은 마스크를 만든다는 문제점이 있다. 

- 경계 박스를 만들기 위해 하나의 네트워크로 객체 박스를 예측하고, 네 개의 추가 네트워크로 아래, 위, 왼쪽 절반, 오른쪽 절반을 예측한다. 

- 불충분한 출력 해상도를 해결하기 위해 DNN localizer를 적용한다. 

### DNN training 

- 이 네트워크의 특징은 단순성이다. classifier는 단순히 mask 생성 레이어로 대체된다. 하지만, 많은 양의 훈련 데이터를 이용한 훈련이 필요하고, 모든 위치에서 크기가 다른 객체의 발생이 필요하다. 

- localization을 위한 훈련은 분류보다 어렵기 때문에 높은 품질의 낮은 필터의 모델 가중치로부터 시작하는 것이 중요하다. 이를 위해 먼저 분류를 위해 네트워크를 훈련시키고, localization을 위해 classifier를 제외한 레이어의 가중치를 다시 사용했다.
